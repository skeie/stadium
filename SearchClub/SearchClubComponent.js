// @flow
import React from 'react';
import { TextInput, View, StyleSheet, TouchableOpacity, ScrollView } from 'react-native';
import Modal from 'react-native-modal';
import { MonoText, Poppins } from '../components/StyledText';
import colors from '../constants/Colors';
import Loading from '../components/Loading';

const styles = StyleSheet.create({
  modalContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.primary,
    padding: 20,
  },
});

type Footballclub = {
  capacity: string,
  name: string,
  stadiumName: string,
};

type FootballclubResult = Array<Footballclub>;

type Props = {
  isVisible: boolean,
  text: string,
  onHandleTextInput: (text: string) => *,
  getMatchData: (club: Footballclub) => *,
  footballclubResult: ?FootballclubResult,
  fetching?: boolean,
  toggleModal: () => void,
  title?: string,
  subTitle?: string,
};

const SearchClub = ({
  isVisible,
  onHandleTextInput,
  footballclubResult,
  getMatchData,
  fetching,
  toggleModal,
  title = 'Oops, there was a problem with that image',
  subTitle = 'Please search and then select the home team in the image.',
}: Props) => {
  return (
    <Modal isVisible={isVisible} onBackdropPress={toggleModal}>
      <View style={styles.modalContainer}>
        <Poppins style={{ marginBottom: 20 }} type="header">
          {title}
        </Poppins>
        <Poppins style={{ width: '100%' }}>{subTitle}</Poppins>
        <TextInput
          width="100%"
          height={50}
          autoFocus
          placeholder="Football club name"
          onChangeText={onHandleTextInput}
        />

        <Poppins width="100%" type="subHeader">
          Result:
        </Poppins>
        {fetching && (
          <View paddingVertical={20}>
            <Loading />
          </View>
        )}
        {footballclubResult && (
          <ScrollView style={{ width: '100%' }} keyboardShouldPersistTaps="handled">
            {footballclubResult.map((result, index) => (
              <TouchableOpacity
                activeOpacity={0.6}
                key={`${result.name}-${index}`}
                onPress={() => {
                  getMatchData(result);
                }}
              >
                <View
                  flexDirection="row"
                  alignItems="center"
                  width="100%"
                  justifyContent="space-between"
                >
                  <View marginVertical={10} width="60%">
                    <View flexDirection="row">
                      <Poppins type="bold">Club: </Poppins>
                      <Poppins>{result.name}</Poppins>
                    </View>
                    <View flexDirection="row">
                      <Poppins type="bold">Stadium: </Poppins>
                      <Poppins numberOfLines={1}>{result.stadiumName}</Poppins>
                    </View>
                  </View>
                  <Poppins
                    style={{
                      borderRadius: 5,
                      padding: 5,
                      borderWidth: 1,
                      borderColor: colors.primaryText,
                      justifyContent: 'center',
                      alignItems: 'center',
                    }}
                  >
                    Select
                  </Poppins>
                </View>
              </TouchableOpacity>
            ))}
          </ScrollView>
        )}
      </View>
    </Modal>
  );
};

export default SearchClub;
