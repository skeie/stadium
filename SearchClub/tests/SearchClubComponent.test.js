
import 'react-native';
import React from 'react';
import SearchClubComponent from '../SearchClubComponent';
import renderer from 'react-test-renderer';

const searchRes = [
    { name: "Real Madrid Castilla", stadiumName: "Alfredo di Stéfano Stadium", capacity: 24000, __typename: "Club" },
    { name: "Real Madrid Club de Fútbol", stadiumName: "Santiago Bernabéu Stadium", capacity: 99454, __typename: "Club" },
    { name: "Real Madrid FC", stadiumName: "Juan Ramón Loubriel Stadium", capacity: 22000, __typename: "Club" },
];


it('Button renders correctly', () => {
    const tree = renderer.create(
        <SearchClubComponent
            getMatchData={jest.fn()} fetching={false}
            toggleModal={jest.fn()}
            isVisible onHandleTextInput={jest.fn()}
            footballclubResult={searchRes}
        />).toJSON();
    expect(tree).toMatchSnapshot();
});

