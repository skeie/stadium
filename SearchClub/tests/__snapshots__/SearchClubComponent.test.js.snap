// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  onModalHide={[Function]}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
    testID={undefined}
  />
  <View
    collapsable={undefined}
    onModalHide={[Function]}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "transform": Array [
          Object {
            "translateY": 1334,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ADF1D2",
          "justifyContent": "center",
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
              "fontSize": 20,
            },
            Object {
              "marginBottom": 20,
            },
          ]
        }
        type="header"
      >
        Oops, there was a problem with that image
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            undefined,
            Object {
              "width": "100%",
            },
          ]
        }
      >
        Please search and then select the home team in the image.
      </Text>
      <TextInput
        allowFontScaling={true}
        autoFocus={true}
        height={50}
        onChangeText={[Function]}
        placeholder="Football club name"
        width="100%"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
            },
            Object {
              "color": "#553555",
              "fontFamily": "poppins",
              "fontSize": 18,
            },
            undefined,
          ]
        }
        type="subHeader"
        width="100%"
      >
        Result:
      </Text>
      <RCTScrollView
        keyboardShouldPersistTaps="handled"
        style={
          Object {
            "width": "100%",
          }
        }
      >
        <View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              alignItems="center"
              flexDirection="row"
              justifyContent="space-between"
              width="100%"
            >
              <View
                marginVertical={10}
                width="60%"
              >
                <View
                  flexDirection="row"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins-bold",
                        },
                        undefined,
                      ]
                    }
                    type="bold"
                  >
                    Club: 
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Real Madrid Castilla
                  </Text>
                </View>
                <View
                  flexDirection="row"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins-bold",
                        },
                        undefined,
                      ]
                    }
                    type="bold"
                  >
                    Stadium: 
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Alfredo di St√©fano Stadium
                  </Text>
                </View>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#553555",
                      "fontFamily": "poppins",
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "borderColor": "#553555",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "justifyContent": "center",
                      "padding": 5,
                    },
                  ]
                }
              >
                Select
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              alignItems="center"
              flexDirection="row"
              justifyContent="space-between"
              width="100%"
            >
              <View
                marginVertical={10}
                width="60%"
              >
                <View
                  flexDirection="row"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins-bold",
                        },
                        undefined,
                      ]
                    }
                    type="bold"
                  >
                    Club: 
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Real Madrid Club de F√∫tbol
                  </Text>
                </View>
                <View
                  flexDirection="row"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins-bold",
                        },
                        undefined,
                      ]
                    }
                    type="bold"
                  >
                    Stadium: 
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Santiago Bernab√©u Stadium
                  </Text>
                </View>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#553555",
                      "fontFamily": "poppins",
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "borderColor": "#553555",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "justifyContent": "center",
                      "padding": 5,
                    },
                  ]
                }
              >
                Select
              </Text>
            </View>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <View
              alignItems="center"
              flexDirection="row"
              justifyContent="space-between"
              width="100%"
            >
              <View
                marginVertical={10}
                width="60%"
              >
                <View
                  flexDirection="row"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins-bold",
                        },
                        undefined,
                      ]
                    }
                    type="bold"
                  >
                    Club: 
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Real Madrid FC
                  </Text>
                </View>
                <View
                  flexDirection="row"
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins-bold",
                        },
                        undefined,
                      ]
                    }
                    type="bold"
                  >
                    Stadium: 
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    numberOfLines={1}
                    style={
                      Array [
                        Object {
                          "color": "#553555",
                          "fontFamily": "poppins",
                        },
                        undefined,
                        undefined,
                      ]
                    }
                  >
                    Juan Ram√≥n Loubriel Stadium
                  </Text>
                </View>
              </View>
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#553555",
                      "fontFamily": "poppins",
                    },
                    undefined,
                    Object {
                      "alignItems": "center",
                      "borderColor": "#553555",
                      "borderRadius": 5,
                      "borderWidth": 1,
                      "justifyContent": "center",
                      "padding": 5,
                    },
                  ]
                }
              >
                Select
              </Text>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</Modal>
`;
